Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py

```diff
@@ -544,17 +554,24 @@ def get_mitre_value_from_id(client, args):
         collection_data = Collection(collection_url, verify=client.verify, proxies=client.proxies)
 
         tc_source = TAXIICollectionSource(collection_data)
-        attack_pattern_obj = tc_source.query([
-            Filter("external_references.external_id", "=", attack_id),
+        attack_pattern_objects = tc_source.query(query=[
             Filter("type", "=", "attack-pattern")
         ])
-        attack_pattern_name = attack_pattern_obj[0]['name'] if attack_pattern_obj else None
+        if attack_pattern_objects:
+            attack_pattern = list(filter(lambda attack_pattern_obj:
+                                  filter_attack_pattern_object_by_attack_id(attack_id,
+                                                                            attack_pattern_obj), attack_pattern_objects))
+            attack_pattern_name = attack_pattern[0]['name']
 
         if attack_pattern_name and len(attack_id) > 5:  # sub-technique
-            parent_name = tc_source.query([
-                Filter("external_references.external_id", "=", attack_id[:5]),
+            parent_objects = tc_source.query([
                 Filter("type", "=", "attack-pattern")
-            ])[0]['name']
+            ])
+            parent_object = list(filter(lambda attack_pattern_obj:
+                                 filter_attack_pattern_object_by_attack_id(attack_id[:5],
```

Comment:
why :5? lets comment the reason/ extract 5 to a constant with a good name

Reply:
if there is a sub-technique, the attack_id will be `T1070.008` , will add a meaningful name